src/main/scala/de/heikoseeberger/sbtheader/HeaderPlugin.scala

```diff
@@ -89,8 +89,13 @@ object HeaderPlugin extends AutoPlugin {
         }(breakOut)
     }
 
-    val headers: SettingKey[Map[String, (Regex, String)]] =
-      settingKey("Header pattern and text by extension; empty by default")
+    val headerLicense: SettingKey[License] =
+      settingKey("The license to apply to files")
```

Comment:
@hseeberger I think we're not quite there. If we change the code like a propose, users can only use our build in Licenses. Any opinion on how to continue allowing custom licenses? I see the following options:

* Change definition to `val headerLicense: SettingKey[String]`. This would require users to call it in their build scripts like so: `headerLicense = Apache2_0("2017", "Heiko Seeberger").text` or `headerLicense = "My custom License text"`
* Change definition to `val headerLicense: SettingKey[Either[License, String]]`. This would require users to call it in their build scripts like so: `headerLicense = Right(Apache2_0("2017", "Heiko Seeberger"))` or `headerLicense = Left("My custom License text")`
* Change the License trait to be no longer sealed. This would require users to call it in their build scripts like so: `headerLicense = new License { override val text = "My custom License text" }`
* Add an additional settingsKey `val headerLicenseText: SettingKey[String]`. So users could chose whether they want to set `headerLicense` or `headerLicenseText`. This would require us to implement some error handling when a user erroneously specifies both settings.

None of my proposals feels really satisfactory. Do you have a better idea?